middlewares/ip_whitelister_test.go

```diff
@@ -0,0 +1,302 @@
+package middlewares
+
+import (
+	"errors"
+	"fmt"
+	"github.com/codegangsta/negroni"
+	"github.com/stretchr/testify/assert"
+	"net"
+	"net/http"
+	"net/http/httptest"
+	"reflect"
+	"testing"
+)
+
+func TestNewIPWhitelister(t *testing.T) {
+	cases := map[string]struct {
```

Comment:
Just to add a bit more reasoning on my end: I agree with you that a map could make sense if every test had a name. However, the parameter you pass to `t.Run()` does not have to be static (as in our case) but could also be the result of some "dynamic computation". For instance, the tests for a math utility function could use the input parameters to define the description.

Looking at it from this angle, employing maps where a static name is needed and a slice of structs where it's not would lead to an inconsistent picture. I prefer tests to look boringly similar. :-)